From 7904e84c1e4a24c81a79dd736d88e98c1e872288 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Fri, 22 Aug 2014 16:24:26 +0100
Subject: [PATCH 305/351] ASoC: florida: Destroy mutex on error path

If we failed snd_soc_register_codec the mutexes were not destroyed. This
patch corrects this.

Change-Id: I6832ec7df70a1a5e4b963aa1f1f9c7c08bec9e56
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 216a98d..40540cc 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -2313,6 +2313,7 @@ static int florida_probe(struct platform_device *pdev)
 			"Failed to register codec: %d\n",
 			ret);
 		snd_soc_unregister_platform(&pdev->dev);
+		goto error;
 	}
 
 	return ret;
-- 
1.7.9.5

